relay-general/src/protocol/thread.rs

```diff
@@ -69,6 +71,114 @@ impl Empty for ThreadId {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "jsonschema", derive(schemars::JsonSchema))]
+#[cfg_attr(feature = "jsonschema", schemars(rename_all = "lowercase"))]
+pub enum LockReasonType {
+    Locked = 1,
+    Waiting = 2,
+    Sleeping = 4,
+    Blocked = 8,
```

Comment:
```suggestion
    Blocked = 8,
    // This enum does not have a `fallback_variant` because we consider it unlikely to be extended. If it is,
    // The error added to `Meta` will tell us to update this enum.
```